Makefile

```diff
@@ -334,6 +341,27 @@ input_cleanup: ## deletes the secret/CM, used by the services as input
 	oc kustomize ${OUT}/${NAMESPACE}/input | oc delete --ignore-not-found=true -f -
 	${CLEANUP_DIR_CMD} ${OUT}/${NAMESPACE}/input
 
+##@ CRC BMO SETUP
+.PHONY: crc_bmo_setup
+crc_bmo_setup:
+	$(eval $(call vars,$@))
+	${CLEANUP_DIR_CMD} ${OPERATOR_BASE_DIR}/baremetal-operator
+	mkdir -p ${OPERATOR_BASE_DIR}
+	oc apply -f ${CERTMANAGER_URL}
+	pushd ${OPERATOR_BASE_DIR} && git clone ${GIT_CLONE_OPTS} $(if $(BMO_BRANCH),-b ${BMO_BRANCH}) ${BMO_REPO} "baremetal-operator" && popd
+	pushd ${OPERATOR_BASE_DIR}/baremetal-operator && sed -i 's/eth2/${PROVISIONING_INTERFACE}/g' ironic-deployment/default/ironic_bmo_configmap.env && popd
+	pushd ${OPERATOR_BASE_DIR}/baremetal-operator && make generate manifests && bash tools/deploy.sh -b -i && popd
```

Comment:
I see a failure locally in my env trying to use this:
```
$ make crc_bmo_setup 
rm -Rf /home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator
mkdir -p /home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/operator
oc apply -f https://github.com/jetstack/cert-manager/releases/download/v1.5.4/cert-manager.yaml
customresourcedefinition.apiextensions.k8s.io/certificaterequests.cert-manager.io unchanged
customresourcedefinition.apiextensions.k8s.io/certificates.cert-manager.io unchanged
customresourcedefinition.apiextensions.k8s.io/challenges.acme.cert-manager.io unchanged
customresourcedefinition.apiextensions.k8s.io/clusterissuers.cert-manager.io unchanged
customresourcedefinition.apiextensions.k8s.io/issuers.cert-manager.io unchanged
customresourcedefinition.apiextensions.k8s.io/orders.acme.cert-manager.io unchanged
namespace/cert-manager unchanged
serviceaccount/cert-manager-cainjector unchanged
serviceaccount/cert-manager unchanged
serviceaccount/cert-manager-webhook unchanged
clusterrole.rbac.authorization.k8s.io/cert-manager-cainjector unchanged
clusterrole.rbac.authorization.k8s.io/cert-manager-controller-issuers unchanged
clusterrole.rbac.authorization.k8s.io/cert-manager-controller-clusterissuers unchanged
clusterrole.rbac.authorization.k8s.io/cert-manager-controller-certificates unchanged
clusterrole.rbac.authorization.k8s.io/cert-manager-controller-orders unchanged
clusterrole.rbac.authorization.k8s.io/cert-manager-controller-challenges unchanged
clusterrole.rbac.authorization.k8s.io/cert-manager-controller-ingress-shim unchanged
clusterrole.rbac.authorization.k8s.io/cert-manager-view unchanged
clusterrole.rbac.authorization.k8s.io/cert-manager-edit unchanged
clusterrole.rbac.authorization.k8s.io/cert-manager-controller-approve:cert-manager-io unchanged
clusterrole.rbac.authorization.k8s.io/cert-manager-controller-certificatesigningrequests unchanged
clusterrole.rbac.authorization.k8s.io/cert-manager-webhook:subjectaccessreviews unchanged
clusterrolebinding.rbac.authorization.k8s.io/cert-manager-cainjector unchanged
clusterrolebinding.rbac.authorization.k8s.io/cert-manager-controller-issuers unchanged
clusterrolebinding.rbac.authorization.k8s.io/cert-manager-controller-clusterissuers unchanged
clusterrolebinding.rbac.authorization.k8s.io/cert-manager-controller-certificates unchanged
clusterrolebinding.rbac.authorization.k8s.io/cert-manager-controller-orders unchanged
clusterrolebinding.rbac.authorization.k8s.io/cert-manager-controller-challenges unchanged
clusterrolebinding.rbac.authorization.k8s.io/cert-manager-controller-ingress-shim unchanged
clusterrolebinding.rbac.authorization.k8s.io/cert-manager-controller-approve:cert-manager-io unchanged
clusterrolebinding.rbac.authorization.k8s.io/cert-manager-controller-certificatesigningrequests unchanged
clusterrolebinding.rbac.authorization.k8s.io/cert-manager-webhook:subjectaccessreviews configured
role.rbac.authorization.k8s.io/cert-manager-cainjector:leaderelection unchanged
role.rbac.authorization.k8s.io/cert-manager:leaderelection unchanged
role.rbac.authorization.k8s.io/cert-manager-webhook:dynamic-serving unchanged
rolebinding.rbac.authorization.k8s.io/cert-manager-cainjector:leaderelection unchanged
rolebinding.rbac.authorization.k8s.io/cert-manager:leaderelection configured
rolebinding.rbac.authorization.k8s.io/cert-manager-webhook:dynamic-serving configured
service/cert-manager unchanged
service/cert-manager-webhook unchanged
deployment.apps/cert-manager-cainjector unchanged
deployment.apps/cert-manager unchanged
deployment.apps/cert-manager-webhook unchanged
mutatingwebhookconfiguration.admissionregistration.k8s.io/cert-manager-webhook configured
validatingwebhookconfiguration.admissionregistration.k8s.io/cert-manager-webhook configured
pushd /home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/operator && git clone  -b main https://github.com/metal3-io/baremetal-operator "baremetal-operator" && popd
~/upstream/git/openstack-k8s-operators/install_yamls/out/operator ~/upstream/git/openstack-k8s-operators/install_yamls
Cloning into 'baremetal-operator'...
remote: Enumerating objects: 45354, done.
remote: Counting objects: 100% (7569/7569), done.
remote: Compressing objects: 100% (1871/1871), done.
remote: Total 45354 (delta 6047), reused 5730 (delta 5693), pack-reused 37785
Receiving objects: 100% (45354/45354), 60.05 MiB | 24.09 MiB/s, done.
Resolving deltas: 100% (19730/19730), done.
~/upstream/git/openstack-k8s-operators/install_yamls
pushd /home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator && sed -i 's/eth2/enp2s0/g' ironic-deployment/default/ironic_bmo_configmap.env && popd
~/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator ~/upstream/git/openstack-k8s-operators/install_yamls
~/upstream/git/openstack-k8s-operators/install_yamls
pushd /home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator && make generate manifests && bash tools/deploy.sh -b -i && popd
~/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator ~/upstream/git/openstack-k8s-operators/install_yamls
make[1]: Entering directory '/home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator'
cd hack/tools; go build -o /home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator/tools/bin/controller-gen sigs.k8s.io/controller-tools/cmd/controller-gen
cd apis; /home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator/tools/bin/controller-gen object:headerFile="../hack/boilerplate.go.txt" paths="./..."
tools/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
cd apis; /home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator/tools/bin/controller-gen "crd:allowDangerousTypes=true,crdVersions=v1" rbac:roleName=manager-role webhook paths="./..." output:webhook:dir=../config/webhook/ output:crd:artifacts:config=../config/crd/bases
tools/bin/controller-gen rbac:roleName=manager-role paths="./..." output:rbac:artifacts:config=config/rbac
cd hack/tools; go build -o /home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator/tools/bin/kustomize sigs.k8s.io/kustomize/kustomize/v4
tools/bin/kustomize build config/default > config/render/capm3.yaml
make[1]: Leaving directory '/home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator'
make[1]: Entering directory '/home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator'
make[1]: Nothing to be done for '/home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator/tools/bin/kustomize'.
make[1]: Leaving directory '/home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator'
~/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator/ironic-deployment/overlays/temp ~/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator
~/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator
~/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator ~/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator
namespace/baremetal-operator-system unchanged
customresourcedefinition.apiextensions.k8s.io/baremetalhosts.metal3.io unchanged
customresourcedefinition.apiextensions.k8s.io/bmceventsubscriptions.metal3.io unchanged
customresourcedefinition.apiextensions.k8s.io/firmwareschemas.metal3.io unchanged
customresourcedefinition.apiextensions.k8s.io/hardwaredata.metal3.io unchanged
customresourcedefinition.apiextensions.k8s.io/hostfirmwaresettings.metal3.io unchanged
customresourcedefinition.apiextensions.k8s.io/preprovisioningimages.metal3.io unchanged
serviceaccount/baremetal-operator-controller-manager unchanged
role.rbac.authorization.k8s.io/baremetal-operator-leader-election-role unchanged
clusterrole.rbac.authorization.k8s.io/baremetal-operator-manager-role configured
clusterrole.rbac.authorization.k8s.io/baremetal-operator-metrics-reader unchanged
clusterrole.rbac.authorization.k8s.io/baremetal-operator-proxy-role unchanged
rolebinding.rbac.authorization.k8s.io/baremetal-operator-leader-election-rolebinding unchanged
clusterrolebinding.rbac.authorization.k8s.io/baremetal-operator-manager-rolebinding unchanged
clusterrolebinding.rbac.authorization.k8s.io/baremetal-operator-proxy-rolebinding unchanged
configmap/baremetal-operator-ironic unchanged
configmap/baremetal-operator-manager-config unchanged
secret/ironic-credentials-72ckkbd957 unchanged
secret/ironic-inspector-credentials-44tcm87mhm unchanged
service/baremetal-operator-controller-manager-metrics-service unchanged
service/baremetal-operator-webhook-service unchanged
deployment.apps/baremetal-operator-controller-manager unchanged
certificate.cert-manager.io/baremetal-operator-serving-cert unchanged
issuer.cert-manager.io/baremetal-operator-selfsigned-issuer unchanged
validatingwebhookconfiguration.admissionregistration.k8s.io/baremetal-operator-validating-webhook-configuration configured
~/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator
~/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator/ironic-deployment/overlays/temp ~/upstream/git/openstack-k8s-operators/install_yamls/out/operator/baremetal-operator
tools/deploy.sh: line 238: IRONIC_HOST_IP: unbound variable
make: *** [Makefile:353: crc_bmo_setup] Error 1
```

Also I had to install apache2-utils package on my host as tools/deploy.sh depends on it. Maybe it worth to add that to https://github.com/openstack-k8s-operators/install_yamls/blob/master/devsetup/roles/download_tools/tasks/main.yaml

Reply:
thanks for the fix for this. The target now works for me locally.